### PR TITLE
Adds the Hunter's Curse

### PR DESCRIPTION
**Pull request by Ms-Mee**
## About The Pull Request

This PR adds a new item opposite to the healing rod of ascelpius, a deal most wicked for those who lust for blood, currently pretty far from done but I'm getting there, based off the crusher. Current progress is as follows:

- [x] Implement the base item.
- [x] Implement the tracking system.
- [x] Implement mark award system.
- [x] Implement the marks.
- [x] Implement the binding.
- [ ] Implement the special effects.
- [ ] Add sprites and sounds.
- [x] Settle on a name and properly flavor it.
- [ ] Clean up the code.

## Why It's Good For The Game

The rod of ascelpius does not directly help hunting very well, you need someone else to use it if you want to hunt, and recent loot rework removed another wicked item that substitutes your arm, the dark blessing, from the mining pool, as such this PR adds something with similar flavor to the dark blessing and quite opposite to the rod of ascelpius.

## Changelog
:cl:
add: A new necropolis chest weapon, similar to the crusher, that binds to one of your arms and rewards you the more you kill.
add: A new item type with variable means of tracking damage dealt, enemies killed and more.
imageadd: There will be new icons for this item, placeholder
/:cl:

### CODE REVIEW

**Comment by HighPrincessErinys** [comment]
asclepius is the greek god of medicine, so why not name this after a greek god of war? ares, or some random obscure one.

**Comment by Ms-Mee** [comment]
Well, it's flavor is currently more towards the blood cult

**Comment by HighPrincessErinys** [comment]
yeah but bloodcult fucking blows ass and sucks and noone likes it we just use cult assets for ruins cause they're spooky noone in the modern age actually likes playing bloodcult

File: code/datums/status_effects/buffs.dm [hunk 1]
```diff
@@ -358,6 +358,82 @@
 	. = ..()
 	STOP_PROCESSING(SSprocessing, src)
 
+//Hippocratic Oath: Applied when the Blood Blessing is activated.
+/datum/status_effect/huntersOath
```
Comment (ZephyrTFA) [inline code/datums/status_effects/buffs.dm]:
```suggestion
/datum/status_effect/hunters_oath
```

;)

Comment (Ms-Mee) [reply code/datums/status_effects/buffs.dm]: It's a counterpart to the rod of asclepius so I'm using the same path naming scheme for the effect, unless you suggest both be changed

Comment (ZephyrTFA) [reply code/datums/status_effects/buffs.dm]: If you want to do that please do

**Comment by Ms-Mee** [comment]
This should work better and fix an issue I was having, key word should
I hope it does.

**Review by AsciiSquid — COMMENTED** [review]
Fear the old blood.

**Comment by Ms-Mee** [comment]
Aaaanyways only 4 marks are yet to be properly implemented: The bear which gives a weaker blood drunk, the ancient and broodmother ones that need effect changes and sprites and all that, the legionnaire one which is in it's effect description and the wolf one which I still don't know yet
No sprites, no sounds, still gotta fix some text.

**Comment by Ms-Mee** [comment]
Checks passed, goood now time to test shit again
It's kinda late but I should lest I forget

**Comment by Ms-Mee** [comment]
Juust 2 marks, a effect redo which needs sounds and sprites, sounds and sprites, the fixing text and shit, this PR will be the end of me

File: code/modules/mining/lavaland/necropolis_chests.dm [hunk 2]
```diff
@@ -1882,13 +1882,24 @@
...
+	throwforce = 10
+	throw_speed = 6
+	armour_penetration = 10
+	custom_materials = list(/datum/material/iron=18000)
+	hitsound = 'sound/weapons/bladeslice.ogg'
+	attack_verb = list("stabbed", "slashed", "cut")
+	sharpness = IS_SHARP_ACCURATE
 	var/activated = FALSE
-	var/usedHand
+	var/targetHand
```
Comment (ZephyrTFA) [inline code/modules/mining/lavaland/necropolis_chests.dm]:
```suggestion
	var/target_hand
```

File: code/modules/mining/lavaland/necropolis_chests.dm [hunk 3]
```diff
@@ -1898,46 +1909,65 @@
 		to_chat(user, "<span class='warning'>The dark being refuses you!</span>")
 		return
 	var/mob/living/carbon/itemUser = user
```
Comment (ZephyrTFA) [inline code/modules/mining/lavaland/necropolis_chests.dm]:
```suggestion
	var/mob/living/carbon/item_user = user
```

**Comment by Ms-Mee** [comment]
Beautiful unfinished unsorted experimental code, I cry every time

**Comment by Ms-Mee** [comment]
I forgor the mob and to remove something
FUN!

**Comment by Ms-Mee** [comment]
Gotta love the CI failures due to map shit

**Comment by Ms-Mee** [comment]
What's CI you might ask? Check Ioure* mom dumbass

**Review by Ms-Mee — COMMENTED** [review]
Looking over everything again it seems nothing is missing other than a proper place for tracked items, which I made since due to some possible intended changes to how loot works having such a system in place is convenient, I'll in a bit make a list with everything about the item itself, it's progression and all that.

**Comment by Ms-Mee** [comment]
Item stats as of current:
Damage: 15 + 10 + 10 + 0.2 per soul up to +10, totalling 45
AP: 10
Sharp
Grants you a medhud, binds to your arm, cannot coexist with Rod of Asclepius.

Marks:
Goliath: Makes the projectile deal 2.5 damage
Ancient Goliath: 5 instead of 2.5, also has from 50 to 100% chance to attack again on mark detonation, higher the more you are hurt
Goliath Broodmother: 7.5 instead of 5, guaranteed extra attack

Watcher: Same as regular crusher
Ice demon(watcher upgrade): same as icewing for regular crusher
Icewing watcher: bonus value of 10

Legion: Grants more damage and lower recharge time the more health you are missing
Dwarf Legion: Bonus value of 2, can also spawn up to the same amount in legion skulls, the more you are hurt
Legionnaire: Bonus value of 3
Legion Megafauna: Bonus value of 4

Polar bear: Damage reduction of 50% on mark detonation, for 1 second
Warbear: 70%
Blood-drunk miner: 90%

Wolf: Just a 3 second speed boost on mark detonation, not much, uses MemedHams' crusher effect I couldn't think of anything better
Alpha Wolf: 5 seconds

Magmawing Watcher: Same as the crusher drake effect but in a smaller radius
Ice whelp: Same radius as regular crusher drake
Drake: bigger radius instead.

Bubblegum: Raises base damage by 5 and increases attack speed by 25%

Colossus: Heals you when you damage things similar to the regular crusher's bubblegum

Hierophant: Same as regular crusher

Monarch's Mark: At 1000 souls, you get more damage and instant projectile recharge, enjoy!

Mark awarding:
You get marks after either getting the right numbers of respective souls in their progression OR killing it's elite or rare mob equivalent, only non-passive non-minion mining mobs give souls. there's also the following soul multipliers:
2 for ice demons and whelps
3 for rare mobs
5 for elites
10 for megafauna

They follow the progression of 10, 30, 50 souls for the regular upgrading order per mob type, and the megafauna ones are awarded at their respective kills or the tresholds of 100, 200, 300, 350, 400 and 500 souls exclusively, with the exception of the ash drake one which can also be attained from 50 flaming souls(from whelps and magmawings)

There is a multiplier which applies to all mark reward tresholds.

The megafauna mark tresholds also give a block chance equal to the base treshold/10, so 10, 20, 30, 35, 40 and 50.

And I believe this is all in and functional, unless I missed something.

**Comment by ZephyrTFA** [comment]
I am going to brutally dissect atmospherics I swear to Derg

**Comment by Ms-Mee** [comment]
As I've stated as much as the code part of this is done I don't particularly feel like finishing it up, if any parts of this are needed that can be done easily.

**Pull request closed without merging**